src/main/kotlin/com/terraformation/backend/search/SearchService.kt

```diff
@@ -176,13 +177,23 @@ class SearchService(private val dslContext: DSLContext) {
 
     val searchResults =
         runQuery(
-            rootPrefix,
-            listOf(fieldPath),
-            criteria,
-            listOf(SearchSortField(fieldPath)),
-            limit = limit,
-            distinct = true,
-        )
+                rootPrefix,
+                listOf(fieldPath),
+                criteria.toExactSearch(),
+                listOf(SearchSortField(fieldPath)),
+                limit = limit,
+                distinct = true,
+            )
+            .ifEmpty {
```

Comment:
Will this effectively run the search twice for a fuzzy query that returns empty?

Reply:
Good catch. I should only do this if `toExactSearch()` makes changes.